src/wp-includes/class-wp-theme.php

```diff
@@ -287,35 +295,39 @@ public function __construct( $theme_dir, $theme_root, $_child = null ) {
 			} else {
 				$this->errors = new WP_Error( 'theme_no_stylesheet', __( 'Stylesheet is missing.' ) );
 			}
-			$this->template    = $this->stylesheet;
-			$this->block_theme = false;
+			$this->template               = $this->stylesheet;
+			$this->block_theme            = false;
+			$this->block_template_folders = $this->get_block_template_folders( true ); // set to default value.
 			$this->cache_add(
 				'theme',
 				array(
-					'block_theme' => $this->block_theme,
-					'headers'     => $this->headers,
-					'errors'      => $this->errors,
-					'stylesheet'  => $this->stylesheet,
-					'template'    => $this->template,
+					'block_template_folders' => $this->block_template_folders,
+					'block_theme'            => $this->block_theme,
+					'headers'                => $this->headers,
+					'errors'                 => $this->errors,
+					'stylesheet'             => $this->stylesheet,
+					'template'               => $this->template,
 				)
 			);
 			if ( ! file_exists( $this->theme_root ) ) { // Don't cache this one.
 				$this->errors->add( 'theme_root_missing', __( '<strong>Error:</strong> The themes directory is either empty or does not exist. Please check your installation.' ) );
 			}
 			return;
 		} elseif ( ! is_readable( $this->theme_root . '/' . $theme_file ) ) {
-			$this->headers['Name'] = $this->stylesheet;
-			$this->errors          = new WP_Error( 'theme_stylesheet_not_readable', __( 'Stylesheet is not readable.' ) );
-			$this->template        = $this->stylesheet;
-			$this->block_theme     = false;
+			$this->headers['Name']        = $this->stylesheet;
+			$this->errors                 = new WP_Error( 'theme_stylesheet_not_readable', __( 'Stylesheet is not readable.' ) );
+			$this->template               = $this->stylesheet;
+			$this->block_theme            = false;
+			$this->block_template_folders = $this->get_block_template_folders( true ); // set to default value.
```

Comment:
This feel odd. Instead of using this, can't we use a class property of `default_template_folders` on this object? Passing a parameter to get these values feel strange

Reply:
@spacedmonkey I was trying to avoid object variable, but it does make sense to have one. I have update the code to added a new class property instead of handling default within function.

Reply:
@kt-12 This class property can be private / protected.